lib/tasks/doctor/check-versions.js

```diff
@@ -0,0 +1,63 @@
+'use strict';
+
+var Task            = require('../../models/task');
+var processVersions = require('../../utilities/get-versions').versions;
+var semver          = require('semver');
+var Table           = require('cli-table');
+var Promise         = require('../../ext/promise');
+var os              = require('os');
+
+module.exports = Task.extend({
+  init: function() {
+    this.npm = this.npm || require('npm');
+    this.os = require('os');
+  },
+  run: function(options) {
+    options = options || {};
+    var mismatches = [];
+    var pkg = this.project.pkg;
+    var userVersions = processVersions();
+
+    if (pkg.os.indexOf(os.platform()) < 0) {
+      mismatches.push(['os', os.platform(), pkg.os.join(',')]);
```

Comment:
seems like we actually get `os` from 3 places.

* this.os
* pkg.os
* var os = require('os')